src/commands/transaction/reconstruct_transaction/mod.rs

```diff
@@ -197,9 +211,25 @@ fn action_transformation(
             )))
         }
         Action::DeployContract(deploy_contract_action) => {
+            // Unfortunately, RPC doesn't return the code for the deployed contract. Only the hash.
+            // So we need to fetch it from archive node.
+
+            let code = crate::commands::contract::download_wasm::get_code(
+                &receiver_id,
+                network_config,
+                block_reference
+            ).map_err(|e| {
+                color_eyre::Report::msg(format!("Couldn't fetch the code. Please use the archival node to fetch the code. Error: {}", e))
```

Comment:
nit: it might be nice to specify in err msg, that archival node should be configured via
`network_connection.*.rpc_url` config field and a link to doc page of all providers added too.